src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,387 @@
+import React from 'react'
+import {Box} from '..'
+import {isResponsiveValue, useResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {SxProp, merge, BetterSystemStyleObject} from '../sx'
+import Heading from '../Heading'
+import {ArrowLeftIcon} from '@primer/octicons-react'
+import Link from '../Link'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
+import {get} from '../constants'
+const REGION_ORDER = {
+  ContextArea: 0,
+  TitleArea: 1,
+  Description: 2,
+  Navigation: 3,
+}
+
+// Types that are shared between sub components
+export type sharedPropTypes = {
+  hidden?: boolean | ResponsiveValue<boolean>
+} & SxProp
+
+// Default state for the `visible` prop when a sub component is only visible on narrow viewport
+const hiddenOnRegularAndWide = {
+  narrow: false,
+  regular: true,
+  wide: true,
+}
+
+// Default state for the `visible` prop when a sub component is visible on regular and wide viewport
+const hiddenOnNarrow = {
+  narrow: true,
+  regular: false,
+  wide: false,
+}
+
+// Root
+// -----------------------------------------------------------------------------
+export type PageHeaderProps = {
+  'aria-label'?: React.AriaAttributes['aria-label']
+  // TODO: check if we need any other types
+  as?: React.ElementType | 'header' | 'div'
+} & sharedPropTypes
+
+const Root: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, as = 'div'}) => {
+  const rootStyles = {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: get('space.2'),
+    // 24px for wide and regular, 16px for narrow viewports
+    padding: useResponsiveValue(
+      {
+        narrow: 3,
+        regular: 4,
+        wide: 4,
+      },
+      false,
+    ),
+  }
+  return (
+    <Box as={as} sx={merge<BetterSystemStyleObject>(rootStyles, sx)}>
+      {children}
+    </Box>
+  )
+}
+
+function displayResponsively<T, F>(value: T, fallback: F): BetterSystemStyleObject {
+  if (isResponsiveValue(value)) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const responsiveValue = value as Extract<T, ResponsiveValue<any>>
+
+    return {
+      [`@media screen and (max-width: 768px)`]: {
+        display: 'narrow' in responsiveValue ? (responsiveValue.narrow ? 'none' : 'flex') : fallback ? 'none' : 'flex',
+      },
+      [`@media screen and (min-width: 768px)`]: {
+        display:
+          'regular' in responsiveValue ? (responsiveValue.regular ? 'none' : 'flex') : fallback ? 'none' : 'flex',
+      },
+      [`@media screen and (min-width: 1440px)`]: {
+        display: 'wide' in responsiveValue ? (responsiveValue.wide ? 'none' : 'flex') : fallback ? 'none' : 'flex',
+      },
```

Comment:
I think this could consider within the [suggestion](https://github.com/primer/react/pull/2265#discussion_r1041511281) as well, so keeping the hard-coded values for now. Let me know if I misunderstand anything!

Reply:
I think you're right - we'll wait until we have the new Primitive tokens.